agent/pkg/proxy/iptables.go

```diff
@@ -143,10 +143,19 @@ func (proxier *Proxier) createIgnoreRules() (ignoreRules []iptablesEnsureInfo, e
 
 	// kube-dns service
 	kubeDNS, err := proxier.kubeClient.CoreV1().Services("kube-system").Get(context.Background(), "kube-dns", metav1.GetOptions{})
-	if err != nil {
-		return
+	if kubeDNS != nil && err == nil {
+		klog.V(4).Infof("ignored kubeDNS: %s", kubeDNS.Name)
+		ignoreRules = append(ignoreRules, ignoreRuleByService(kubeDNS))
+	}
+
+	// coredns service
+	kubeDNSList, err := proxier.kubeClient.CoreV1().Services("kube-system").List(context.Background(), metav1.ListOptions{LabelSelector: "k8s-app=kube-dns"})
```

Comment:
Is the service name of coredns also kube-dns?

Reply:
the coredns service:
```shell
NAME      TYPE        CLUSTER-IP   EXTERNAL-IP   PORT(S)                        AGE     LABELS
coredns   ClusterIP   10.10.0.3    <none>        53/UDP,53/TCP,9153/TCP         4d18h   addonmanager.kubernetes.io/mode=Reconcile,k8s-app=kube-dns,kubernetes.io/name=coredns
```

Reply:
Okay

Reply:
But will this cause us to add the same rule twice?

On my cluster:
```
[root@k8s-master ~]# kubectl get svc -n kube-system
NAME       TYPE        CLUSTER-IP   EXTERNAL-IP   PORT(S)                  AGE
kube-dns   ClusterIP   10.96.0.10   <none>        53/UDP,53/TCP,9153/TCP   16d
```

kubeDNS will add the rule for the first time, and kubeDNSList will add the rule for the second time.

Reply:
In order to keep only one iptables rule, can we delete kubeDNS?

Reply:
It is indeed possible, so I created a map to deal with the problem that multiple retrievals may cause chain duplication.

Reply:
Perfect

Reply:
```suggestion
	kubeDNSList, err := proxier.kubeClient.CoreV1().Services(metav1.NamespaceSystem).List(context.Background(), metav1.ListOptions{LabelSelector: "k8s-app=kube-dns"})
```